diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 21d8eb8274..350a0623cf 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -9260,7 +9260,7 @@ static bool setting_append_list(
               parent_group);
 #endif
 
-#if defined(HAVE_LAKKA)
+#if defined(HAVE_LAKKA) || defined(HAVE_ODROIDGO2)
 #ifdef HAVE_LAKKA_SWITCH
         CONFIG_ACTION(
                list, list_info,
diff --git a/retroarch.c b/retroarch.c
index 0f9151d2d1..07737f907e 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -2450,7 +2450,7 @@ bool command_event(enum event_command cmd, void *data)
          system("(sleep 1 && shutdown -P now) & disown");
 #else
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -P now");
+         system("sudo shutdown -P now");
 #endif /* HAVE_LAKKA */
 #endif
          break;
@@ -2465,7 +2465,7 @@ bool command_event(enum event_command cmd, void *data)
          system("(sleep 1 && shutdown -r now) & disown");
 #else
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -r now");
+         system("sudo shutdown -r now");
 #endif /* HAVE_LAKKA */
 #endif
          break;
